File: statsig-java/src/main/java/com/statsig/DynamicConfig.java
package com.statsig;

import com.alibaba.fastjson2.annotation.JSONCreator;
import com.alibaba.fastjson2.annotation.JSONField;
import java.util.Map;

public class DynamicConfig extends BaseConfig {
  @JSONCreator
  DynamicConfig(
      @JSONField(name = "name") String name,
      @JSONField(name = "value") Map<String, Object> value,
      @JSONField(name = "ruleID") String ruleID,
      @JSONField(name = "details") EvaluationDetails evaluationDetails,
      @JSONField(name = "idType") String idType) {
    super(name, value, ruleID, evaluationDetails, idType);
  }
}

File: statsig-elixir/lib/statsig/dynamic_config.ex
defmodule Statsig.DynamicConfig do
  defstruct [
    :name,
    :value,
    :rule_id,
    :id_type
  ]

  @type t :: %__MODULE__{
          name: String.t(),
          value: %{optional(String.t()) => any()},
          rule_id: String.t(),
          id_type: String.t()
        }
end
